src/modules/login/index.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useReducer, useState } from 'react';
+
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import {
+    Image,
+    ImageBackground,
+    Keyboard,
+    KeyboardAvoidingView,
+    Platform,
+    StyleSheet,
+    View,
+} from 'react-native';
+
+import { Routes } from '@constants/index';
+
+import { Button, FieldWrapper, TextInput } from '@app/components';
+import SocialLine from '@app/modules/login/components/SocialLine';
+import { RootStackParamList } from '@app/navigation/RootStackParamList';
+import { layout } from '@app/styles/layout';
+import { handleReducerChange, stateReducer } from '@app/utils/ReactReducer';
+
+type Props = NativeStackScreenProps<RootStackParamList, Routes.settings>;
+
+const Login = ({ navigation }: Props) => {
+    const initialState = {
+        login: '',
+        password: '',
+    };
+
+    const [errors, errorsDispatch] = useReducer(stateReducer, {});
+    const handleErrorsChange = handleReducerChange(errorsDispatch);
+
+    const [values, valuesDispatch] = useReducer(stateReducer, initialState);
+    const handleValuesChange = handleReducerChange(valuesDispatch);
+
+    const handleInputChange = (field: string) => (value: string) => {
+        handleValuesChange({ [field]: value });
+    };
+
+    const [isSubmitting, setIsSubmitting] = useState(false);
+
+    const handleSubmit = () => {
+        Keyboard.dismiss();
+        handleErrorsChange({});
+        setIsSubmitting(true);
+        console.log(values);
```

Comment:
console.log)

Reply:
Yeah. I don't have any login functionality in this pull request, so I just output values to console to indicate that it's working :)